packages/compass-telemetry/src/generic-track.ts

```diff
@@ -1,52 +1,53 @@
-import { createLogger, mongoLogId } from '@mongodb-js/compass-logging';
+import {
+  type Logger,
+  mongoLogId,
+  createLogger,
+} from '@mongodb-js/compass-logging';
 
-type TrackProps = Record<string, any> | (() => Record<string, any>);
+export type TrackProps = Record<string, any> | (() => Record<string, any>);
 export type TrackFunction = (event: string, properties?: TrackProps) => void;
 
-const { log, debug } = createLogger('COMPASS-TELEMETRY');
-
 export interface TelemetryPreferences {
   getPreferences(): { trackUsageStatistics: boolean };
 }
 
-export const createGenericTrack = (
-  emit: (event: string, arg: any) => void,
-  preferencesService?: TelemetryPreferences
-) => {
+export const createGenericTrack = ({
+  sendTrack,
+  logger,
+  preferences,
+}: {
+  sendTrack: (event: string, properties: TrackProps) => void;
```

Comment:
This type is not correct, properties were already awaited by the internal logic of the tracker here (which is great, this means that we are sharing logic that shouldn't differ between environment)

```suggestion
  sendTrack: (event: string, properties: Record<string, any>) => void;
```